docs/developer/development/ci.rst

```diff
@@ -0,0 +1,146 @@
+Continuous Testing
+==================
+
+Add a new test in the continuous integration system
+---------------------------------------------------
+
+A test as continuous integration system level is an End-to-End test that need a
+dedicated environment (with one or several machines (virtual or container)).
+
+A test that only checks a specific feature of a classic MetalK8s deployment
+should be part of PyTest BDD and not integrated as a dedicated stage in
+continuous integration system.
+
+How to choose between Pre-merge and Post-merge
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The choice really depends on the goals of this test.
+
+As a high-level view:
+
+Pre-merge:
+
+- Test is usually not long and could last less than 30 minutes.
+- Test essential features of the product (installation, expansion, building,
+  ...).
+
+Post-merge:
+
+- Test last longer (more than 30 minutes).
+- Test "non-essential" feature of the product (upgrade, downgrade, solutions,
+  ...).
+
+How to add a stage in continuous integration system
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
Could be worth adding a paragraph to explain how to make some stages optional (using build properties and `doStepIf`), and also remind that it's possible to group related stages under a parent stage, using the `TriggerStages` method.

Reply:
Hmm for me it's part of `eve` documentation, we shouldn't talk about it here (even if I know that opensource world do not have access to eve documentation, I think they will not develop stuff for eve because they do not have any access to eve, build, artifacts, ....)

Reply:
OK. That was more to provide guidance to our own devs, but I get why you'd rather have this in Eve docs directly.